src/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjection.java

```diff
@@ -0,0 +1,494 @@
+	private static final String[][] SLEEP_INJECTION = {
+			// Attacking where and group clauses
+			{"'); sleep("+INSERT_TIME_SHORT+"); print('","'); sleep("+INSERT_TIME_LONG+"); print('"}, 
+			{"'; sleep("+INSERT_TIME_SHORT+"); var x='", "'; sleep("+INSERT_TIME_LONG+"); var x='"},
+			{"; sleep("+INSERT_TIME_SHORT+")", "; sleep("+INSERT_TIME_LONG+")"},
+			{"zap); sleep("+INSERT_TIME_SHORT,"zap); sleep("+INSERT_TIME_LONG },
+			// Attacking mapReduce clause
+			{"_id);}, function(inj) { sleep("+INSERT_TIME_SHORT+");return 1;}, { out: 'x'}); "+
+			"db.injection.mapReduce(function() { emit(1,1",	
+			"_id);}, function(inj) { sleep("+INSERT_TIME_LONG+"); return 1;}, { out: 'x'}); "+
+			"db.injection.mapReduce(function() { emit(1,1"}};
+	private static final String[][] JSON_INJECTION = {{"$ne", "0"}, {"$gt", ""}, {"$regex", ".*"}};
+	// Error messages that addressing to a well-known vulnerability
+	private  final Pattern[] errorPatterns = {
+			Pattern.compile("RuntimeException: SyntaxError: unterminated string literal", Pattern.CASE_INSENSITIVE),
+			Pattern.compile("MongoResultException", Pattern.CASE_INSENSITIVE)};
+	// Log prints
+	private static final String JSON_EX_LOG = "try to convert the payload in json format";
+	private static final String IO_EX_LOG = "try to send an http message";
+	private static final String URI_EX_LOG = "try to get the message's Uri";
+	private static final Logger LOG = Logger.getLogger(MongoDbInjection.class);
+
+	@Override
+	public int getCweId() {
+		return 943;
+	}
+
+	@Override
+	public int getWascId() {
+		return 19;
+	}
+
+	public int getId() {
+		return 40033;
+	}
+
+	public Tech getTech() {
+		//TODO change in Tech.MongoDB as soon as available
+		return Tech.Db;
+	}
+
+	@Override
+	public boolean targets(TechSet technologies) {
+		//TODO change in Tech.MongoDB as soon as available
+		return technologies.includes(Tech.Db);
+	}
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "name");
+	}
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.INJECTION;
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return new String[] {};
+	}
+
+	public String getExtraInfo(String attack) {
+		return Constant.messages.getString(MESSAGE_PREFIX + "extrainfo." + attack);
+	}
+
+	@Override
+	public void init() {
+		if (LOG.isDebugEnabled()) {
+			LOG.debug("Initialising MongoDB penertration tests");
+		}
+		if (this.getAttackStrength() == AttackStrength.LOW) {
+			SLEEP_TIME_SHORT = 3000;
+			SLEEP_TIME_LONG = 5000;
+			doCrashScan = false;
+			doAllDataScan = true;
+			doTimedScan = true;
+			doJsonScan = true;

```

Comment:
It seems like `doAllDataScan `, `doTimedScan`, `doJsonScan`, and `doAuthBypass` are always true. So they should be true from the start? Or perhaps just aren't needed at all?

If a case statement was used here then HIGH could be handled as a fall through to INSANE?

Reply:
I thought about setting up the controls to have a way to change the power of the scans even in the perspective of future additions